components/script/textinput.rs

```diff
@@ -864,20 +988,20 @@ impl<T: ClipboardProvider> TextInput<T> {
     }
 
     /// The length of the content in bytes.
-    pub fn len(&self) -> usize {
-        self.lines.iter().fold(0, |m, l| {
-            m + l.len() + 1 // + 1 for the '\n'
-        }) - 1
+    pub fn len(&self) -> ByteOffset {
+        self.lines.iter().fold(ByteOffset::zero(), |m, l| {
+            m + ByteOffset(l.len()) + ByteOffset::one() // + 1 for the '\n'
+        }) - ByteOffset::one()
     }
 
-    /// The length of the content in bytes.
-    pub fn utf16_len(&self) -> usize {
-        self.lines.iter().fold(0, |m, l| {
-            m + l.chars().map(char::len_utf16).sum::<usize>() + 1 // + 1 for the '\n'
-        }) - 1
+    /// The total number of code units required to encode the content in utf16.
+    pub fn utf16_len(&self) -> UTF16CodeUnits {
+        self.lines.iter().fold(UTF16CodeUnits::zero(), |m, l| {
+            m + UTF16CodeUnits(l.chars().map(char::len_utf16).sum::<usize>() + 1) // + 1 for the '\n'
+        }) - UTF16CodeUnits::one()
     }
 
-    /// The length of the content in chars.
+    /// The length of the content in UTF-8 chars.
```

Comment:
This is not correct. Rust’s `char` is a *Unicode scalar value* (which is close enough to a *Unicode code point* for our purpose here) and is not any more related to UTF-8 than it is to UTF-16. Please change this to `length of the content in Unicode code points`.

Reply:
https://simonsapin.github.io/wtf-8/ has some more background and terminology for these concepts.

Reply:
got it, thanks for the clarification!